### PR TITLE
Redo #160

### LINKED ISSUE #156: TypeError: unsupported operand type(s) for ^: 'float' and 'int
Trying to import AGB 3.1.1 from the SimaproCSV:

```
Traceback (most recent call last):
  File "/home/.../import_agb311.py", line 62, in <module>
    agb_importer = bw.SimaProCSVImporter(agb_csv_filepath, "agribalyse3.1")
  File "/home/.../brightway2-io/bw2io/importers/simapro_csv.py", line 45, in __init__
    self.data, self.global_parameters, self.metadata = SimaProCSVExtractor.extract(
  File "/home/.../brightway2-io/bw2io/extractors/simapro_csv.py", line 120, in extract
    ds, index = cls.read_data_set(
  File "/home/.../brightway2-io/bw2io/extractors/simapro_csv.py", line 514, in read_data_set
    cls.parse_input_line(data[index], category, pm)
  File "/home/.../brightway2-io/bw2io/extractors/simapro_csv.py", line 358, in parse_input_line
    is_formula = not isinstance(to_number(amount), Number)
  File "/home/.../brightway2-io/bw2io/extractors/simapro_csv.py", line 54, in to_number
    return float(eval(obj.replace(",", ".").strip()))
  File "<string>", line 1, in <module>
TypeError: unsupported operand type(s) for ^: 'float' and 'int'
``` 

A debugging session leads to the problem: `10^6` cannot be evaluated by Python's eval:

```
> /home/dadafkas/prelab/ecobalyse/brightway/brightway2-io/bw2io/extractors/simapro_csv.py(59)to_number()->None
-> import pdb; pdb.set_trace()  # fmt: skip
(Pdb) float(eval(obj.replace(",", ".").strip()))
*** TypeError: unsupported operand type(s) for ^: 'float' and 'int'
(Pdb) obj.replace(",", ".")
'(38-15)*4185*30/0.9*10^-6'
```

### PR DESCRIPTION
**Pull request by ccomb**
Fix #156 #159 and redo broken PR #160
Don't pin numpy, evaluate expressions in allocations, convert exponent in simapro csv import

### CODE REVIEW

File: bw2io/extractors/simapro_csv.py
Comment (BenPortner) [inline bw2io/extractors/simapro_csv.py:1]: Hi @ccomb this is not your fault but can I ask you to fix something while we're working on this part? Using Python's native eval is unsave for arbitrary user inputs. Could you change the line to use bw2parameter's `DefaultInterpreter` instead (or `asteval.Interpreter` if Chris does not want to make bw2parameters a mandatory dependency)? That would be very helpful :)

File: bw2io/strategies/simapro.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 import re
 import numpy as np
 from stats_arrays import LognormalUncertainty
+import bw2parameters
```
Comment (BenPortner) [inline bw2io/strategies/simapro.py:18]: This means we have to add bw2parameters to the setup requirements. What do you say @cmutel?

File: bw2io/extractors/simapro_csv.py
Comment (ccomb) [reply bw2io/extractors/simapro_csv.py:1]: Exxxcccellent idea to remove an eval. I'll look at it this week

Comment (ccomb) [reply bw2io/extractors/simapro_csv.py:1]: Are we sure `asteval` is a good choice?  I'll read the doc to see if the parser can be configured
```
>>> asteval.Interpreter().eval("2+2")
4
>>> asteval.Interpreter().eval("2^2")
0
>>> asteval.Interpreter().eval("2**2")
4
>>> asteval.Interpreter().eval("2^-2")
-4
>>> asteval.Interpreter().eval("2**-2")
0
>>> asteval.Interpreter().eval("2.0^-2")
>>> asteval.Interpreter().eval("2.0**-2")
0.25
>>> asteval.Interpreter().eval("2.0^(-2)")
>>> asteval.Interpreter().eval("2.0**(-2)")
0.25
```

Comment (ccomb) [reply bw2io/extractors/simapro_csv.py:1]: If we still convert `^` to `**` in the input data, the only problem is the evaluation of `2**-2`. I've asked as an `asteval` issue here: https://github.com/newville/asteval/issues/115

Comment (ccomb) [reply bw2io/extractors/simapro_csv.py:1]: It's ok to merge, with `eval` replaced with `asteval`.
But we need a fixed version of `asteval`

**Comment by BenPortner** [comment]
Hi @ccomb,

Let's move the eval replacement to a new PR and get everything else merged for now. Can you roll back the eval change and make sure all tests are passing?

Cheers.
Ben

**Comment by ccomb** [comment]
The problem with using only asteval is that the parameter variables are not evaluated and I get a lot of warnings in the console during an ecoinvent import, but I had some troubles with bw2parameter also.  I'll try to find some time in the next two or three weeks to find a solution.  In the meantime I'll split this PR into smaller ones.

**Comment by ccomb** [comment]
I created another PR #201 with just asteval
(But still waiting for asteval 0.9.30 with https://github.com/newville/asteval/pull/116 from @newville)

**Pull request closed without merging**